subsystems/visionsubsystem.py

```diff
@@ -1,76 +1,88 @@
-
-        if not multitagresult.estimatedPose.isPresent:
-            ambiguity = 10
-            for result in photonResult.targets:
-                if result.poseAmbiguity < ambiguity:
-                    bestRelativeTransform = (
-                        Transform3d(
-                            Pose3d(), constants.kApriltagPositionDict[result.fiducialId]
+            if not multitagresult.estimatedPose.isPresent:
+                ambiguity = 10
+                for result in photonResult.targets:
+                    if result.poseAmbiguity < ambiguity:
+                        bestRelativeTransform = (
+                            Transform3d(
+                                Pose3d(),
+                                constants.kApriltagPositionDict[result.fiducialId],
+                            )
+                            + result.bestCameraToTarget.inverse()
                         )
-                        + result.bestCameraToTarget.inverse()
-                    )
-                    ambiguity = result.poseAmbiguity
+                        ambiguity = result.poseAmbiguity
+
+            self.updateAdvantagescopePose(Pose3d() + bestRelativeTransform, camera)
 
-        botPose = (
-            Pose3d()
-            + bestRelativeTransform
-            + constants.kLimelightRelativeToRobotTransform.inverse()
-        )
+            cameraTransforms = {
+                "frontLeft": constants.kRobotToFrontLeftCameraTransform,
+                "frontRight": constants.kRobotToFrontRightCameraTransform,
+                "backLeft": constants.kRobotToBackLeftCameraTransform,
+                "backRight": constants.kRobotToBackRightCameraTransform,
+            }
 
-        self.drive.visionEstimate = botPose.toPose2d()
+            botPose = (
+                Pose3d()
+                + bestRelativeTransform
+                + cameraTransforms[camera.getName()].inverse()
+            )
 
-        SmartDashboard.putBoolean(
-            constants.kRobotVisionPoseArrayKeys.validKey, hasTargets
-        )
-        SmartDashboard.putNumberArray(
-            constants.kRobotVisionPoseArrayKeys.valueKey,
-            [
-                self.drive.visionEstimate.X(),
-                self.drive.visionEstimate.Y(),
-                self.drive.visionEstimate.rotation().radians(),
-            ],
-        )
+            self.poseList.append(
+                EstimatedPose(botPose, hasTargets, photonResult.getTimestamp())
+            )
 
-    def updateAdvantagescopePose(self) -> None:
-        limelightPose3d = (
-            pose3dFrom2d(self.estimatedPosition)
-            + constants.kLimelightRelativeToRobotTransform
-        )
-        limelightPose = advantagescopeconvert.convertToSendablePoses([limelightPose3d])
+    def updateAdvantagescopePose(
+        self, cameraPose3d: Pose3d, camera: PhotonCamera
+    ) -> None:
+        cameraKeys = {
+            "frontLeft": constants.kPhotonvisionFrontLeftCameraKey,
```

Comment:
see above comment with regards to merging this into a larger struct, purpose being to allow a more dynamic number of cameras